File: BE/routes/submission.routes.js
const express = require('express');
const router = express.Router();
const multer = require('multer');
const path = require('path');
const submissionController = require('../controllers/submission.controller');
const { authenticateToken } = require('../middleware/auth');
const { requireRole } = require('../middleware/role');

// Multer config for submission files
const submissionStorage = multer.diskStorage({
  destination: (req, file, cb) => cb(null, path.join(__dirname, '../uploads/submissions')),
  filename: (req, file, cb) => cb(null, Date.now() + '-' + file.originalname)
});
const uploadSubmission = multer({
  storage: submissionStorage,
  fileFilter: (req, file, cb) => {
    const allowed = ['application/pdf', 'application/zip', 'application/x-zip-compressed'];
    if (allowed.includes(file.mimetype)) cb(null, true);
    else cb(new Error('Only PDF and ZIP files allowed'), false);
  },
  limits: { fileSize: 20 * 1024 * 1024 }
});

// Student routes
router.post('/:taskId', authenticateToken, requireRole('student'), uploadSubmission.single('file'), submissionController.submitWork);
router.get('/mine', authenticateToken, requireRole('student'), submissionController.getMySubmissions);

// Company routes
router.get('/company/all', authenticateToken, requireRole('company'), submissionController.getAllCompanySubmissions);
router.get('/task/:taskId', authenticateToken, requireRole('company'), submissionController.getTaskSubmissions);
router.get('/:id', authenticateToken, submissionController.getSubmission);

module.exports = router;

File: BE-Python/routes/submission_routes.py
from flask import Blueprint
from middleware.auth import authenticate_token
from middleware.role import require_role
from controllers import submission_controller

submission_bp = Blueprint('submissions', __name__)

@submission_bp.route('/<int:task_id>', methods=['POST'])
@authenticate_token
@require_role('student')
def submit_work(task_id):
    return submission_controller.submit_work(task_id)

@submission_bp.route('/mine', methods=['GET'])
@authenticate_token
@require_role('student')
def get_my_submissions():
    return submission_controller.get_my_submissions()

@submission_bp.route('/company/all', methods=['GET'])
@authenticate_token
@require_role('company')
def get_all_company_submissions():
    return submission_controller.get_all_company_submissions()

@submission_bp.route('/task/<int:task_id>', methods=['GET'])
@authenticate_token
@require_role('company')
def get_task_submissions(task_id):
    return submission_controller.get_task_submissions(task_id)

@submission_bp.route('/<int:id>', methods=['GET'])
@authenticate_token
def get_submission(id):
    return submission_controller.get_submission(id)
